### ISSUE TITLE
help request: response-rewrite can't get body from core.response.hold_body_chunk(ctx)

### ISSUE THREAD

**Issue by ghost** [apache/apisix#9048]
### Description

I made a plugin,and i return 200,**msgInfo** in plugin before_proxy.

Then i try to use response-rewrite to modify the **msgInfo**

but it failed,then I debug that, the response-rewrite get the body by  core.response.hold_body_chunk(ctx)

and then I found that  the result is nli or ''

I wonder why~

### Environment

- APISIX version (the latest version):

**Comment by An-DJ** [comment]
@Brucepower0223 Could you show ur plugin and the detailed steps to help us reproduce the issue?

**Comment by monkeyDluffy6017** [comment]
Could you log in `core.response.hold_body_chunk` to see if other plugin have used this function?

**Comment by Brucepower0223** [comment]
> Could you log in `core.response.hold_body_chunk` to see if other plugin have used this function?

yes, I can make sure that only my plugin and response-rewreite plugin do work

**Comment by Brucepower0223** [comment]
> @Brucepower0223 Could you show ur plugin and the detailed steps to help us reproduce the issue?

yes.
![image](https://user-images.githubusercontent.com/12574550/224880281-890fc42c-ae49-450d-bac9-6537b7650c2f.png)


I return core.json.encode(msg). in proxy phrase. And i tried use response rewrite to modify this. And i log in response-rewreite and I found. In core.response.hold_body_chunk. 
![image](https://user-images.githubusercontent.com/12574550/224880507-f21a0c5c-b026-4212-8e0f-e1eca8dad021.png)

I logged after      local body = core.response.hold_body_chunk(ctx)
![image](https://user-images.githubusercontent.com/12574550/224880587-bf8bcbbb-b1be-4dfb-9a8f-d1fbf7e954af.png)


the body is nil or ''
